sdk/iot/inc/az_iot_mqtt.h

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <az_span.h>
+
+#ifndef _az_IOT_MQTT_H
+#define _az_IOT_MQTT_H
+
+#include <_az_cfg_prefix.h>
+
+typedef struct az_iot_mqtt_connect {
+    az_span client_id;  // NULL terminated string.
+    az_span user;       // NULL terminated string.
+    az_span password;   // NULL terminated string (may be an empty string).
+} az_iot_mqtt_connect;
+
+typedef struct az_iot_topic {
+    az_span name;       // NULL terminated string.
+    uint8_t qos;
+} az_iot_topic;
+
+typedef struct az_iot_mqtt_publish {

```

Comment:
should this be generalized to `az_iot_mqtt_msg`? We could use this to parse responses into as well.

Reply:
This is already used for receiving: it's a pub message in that case as well.
Offline discussion: do we actually need this?